fits-loader-docker/sqs-process.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+set -e
+
+RAW_MESSAGE=$(aws sqs receive-message --wait-time-seconds 20 --visibility-timeout 300 --queue-url "${SQS_QUEUE_URL}")
+
+if [[ "${RAW_MESSAGE}" != "" ]]; then
+  echo
+  echo "received event from SQS"
+  MESSAGEBODY=$(echo "${RAW_MESSAGE}" | jq --raw-output '.Messages[0].Body')
+  MESSAGERH=$(echo "${RAW_MESSAGE}" | jq --raw-output '.Messages[0].ReceiptHandle')
+  EVENTTYPE=$(echo "${MESSAGEBODY}" | jq --raw-output '.Records[0].eventName')
+
+  if [[ "${EVENTTYPE}" == "ObjectCreated:Put" ]]; then
+    S3PATH=$(echo "${MESSAGEBODY}" | jq --raw-output '"s3://" + .Records[0].s3.bucket.name + "/" + .Records[0].s3.object.key')
+    FILENAME=$(echo "${MESSAGEBODY}" | jq --raw-output '.Records[0].s3.object.key')
+
+    aws s3 cp "${S3PATH}" "${FILENAME}"
+
+    if [ -d data ]; then
+      rm -rf data
+    fi
+    mkdir data
```

Comment:
This is a construct I've used for this sort of thing ... it generally cleans up after the script completes (you can add extra exit codes as expected).
```
trap 'rm -rf "$TMPDIR"' EXIT RETURN
TMPDIR=$(mktemp -d)|| exit 1
```

Reply:
I generally would have this at the start of the script.